### PR TITLE
Add support of int64 and fix 2d bool support in h5wasm

### LINKED ISSUE #1178: [h5wasm] Arrays of 64-bit integers are not supported
### Describe the bug

When loading a file with _h5wasm_, opening a dataset with the dtype `<i8` raises an error `Expected array or typed array` in the Visualize area.

### To Reproduce

1. Go to https://h5web.panosc.eu/h5wasm?url=https%3A%2F%2Fzenodo.org%2Fapi%2Ffiles%2Fb0bfa823-917d-4fe6-9a02-4396743a6037%2Fxrr_dataset.h5
2. Click on the `/DIP_1/experiment/scan` dataset
3. See error `Expected array or typed array`

The same dataset can be visualized fine when served by h5grove: https://h5web.panosc.eu/h5grove?file=xrr_dataset.h5

### Expected behaviour

Data contained in such datasets should be able to be visualized when loaded with h5wasm.

### More

The failing check is here: https://github.com/silx-kit/h5web/blob/d84a50f776b69da88937b2c99e2cc4e3a2cb3c41/packages/shared/src/guards.ts#L122. The failure comes from the fact that h5wasm encodes the values for such datasets in [BigInt64Array](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/BigInt64Array). 

For some reason, `isTypedArrayLodash` returns `false` for these arrays in the browser (while I had `true` when testing in `node`...). Although, even if the check succeeded, we would end up with an array with `BigInt` objects that must be converted to `Number` before feeding them to visualisations. So in any case, a conversion step is needed in `h5wasm-api`. 

So I see two possiblities:
- Do the conversion ourselves in `getValue` by detecting 64-bit integer datasets and mapping the resulting value.
- Let `to_array` from `h5wasm` handle it for such datasets. [It is supposed to return JSON-compatible values](https://github.com/usnistgov/h5wasm/blob/f1d7547255c32c7723dcd5e30e101e100bc7dc2b/CHANGELOG.md#added) even though it is not the case for now (but we could argue that as `BigInt` is converted to `Number`, `BigInt` arrays should be converted to `Number` arrays).

### PR DESCRIPTION
**Pull request by loichuder**
Fix #1178

@linupi This should fix the issue for the file you gave as example in #1158

### CODE REVIEW

File: packages/h5wasm/src/guards.ts [hunk 1]
```diff
@@ -93,3 +95,7 @@ export function assertNumericMetadata(
     throw new Error('Expected H5Wasm numeric metadata');
   }
 }
+
+export function hasInt64Type(dataset: Dataset) {
+  return dataset.type.class === DTypeClass.Integer && dataset.type.size === 64;
+}
```
Comment (loichuder) [inline packages/h5wasm/src/guards.ts:101]: I could put this with other guards in `shared` but it is only useful for h5wasm and we don't need to assert the TS type like the guards in `shared` do.

File: packages/h5wasm/src/h5wasm-api.ts [hunk 2]
```diff
@@ -54,12 +56,15 @@ export class H5WasmApi extends ProviderApi {
     const h5wDataset = await this.getH5WasmEntity(dataset.path);
     assertH5WasmDataset(h5wDataset);
 
-    if (hasBoolType(dataset)) {
-      const value = h5wDataset.to_array();
+    // h5wasm returns integers for bool and BigInt for int64
+    // So we use to_array instead to have bool and numbers resp.
+    if (hasBoolType(dataset) || hasInt64Type(dataset)) {
```
Comment (loichuder) [inline packages/h5wasm/src/h5wasm-api.ts:61]: Two-fold fix:

1. Use `to_array` for `int64` as well to have `BigInt` converted to numbers.

File: packages/h5wasm/src/h5wasm-api.ts [hunk 3]
```diff
@@ -54,12 +56,15 @@ export class H5WasmApi extends ProviderApi {
...
+      const rawValue = h5wDataset.to_array();
+      const value = hasArrayShape(dataset)
+        ? flattenValue(rawValue, dataset)
+        : rawValue;
```
Comment (loichuder) [inline packages/h5wasm/src/h5wasm-api.ts:65]: 2. As `to_array` returns nested arrays for nD datasets and as the visus expect 1D array, flatten it.

The previous code was flattening the result of `to_array` only if there was slicing. That means that 2D arrays were not flattened for 2D+ vis (`Matrix`, `Heatmap`),  which was breaking them.

**Review by axelboc — APPROVED** [review]
👏

**Pull request merged**